WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -215,18 +206,16 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
 
         // Enable the block editor on sites created on mobile
-        switch (requestCode) {
-            case RequestCodes.CREATE_SITE:
-                if (data != null) {
-                    int newSiteLocalID = data.getIntExtra(SitePickerActivity.KEY_LOCAL_ID, -1);
-                    SiteUtils.enableBlockEditorOnSiteCreation(mDispatcher, mSiteStore, newSiteLocalID);
-                    // Mark the site to show the GB popup at first editor run
-                    SiteModel newSiteModel = mSiteStore.getSiteByLocalId(newSiteLocalID);
-                    if (newSiteModel != null) {
-                        AppPrefs.setShowGutenbergInfoPopupForTheNewPosts(newSiteModel.getUrl(), true);
-                    }
+        if (requestCode == RequestCodes.CREATE_SITE) {
```

Comment:
This could be part of the switch about, right? Maybe it would make the code a bit more readable since it's checking the same request code 👍

Reply:
Good catch! I didn't notice that the switch above also checks `RequestCodes.CREATE_SITE` :) One thing though, `ADD_ACCOUNT` and `CREATE_SITE` are lumped together in fallthrough there, so while we can move the code up there,  we will still have to check if it's `RequestCodes.CREATE_SITE`  (at least that's how I see it), which doesn't look too good either :) I think we could leave this part as-is for now - it should be gone when block editor replaces Aztec.

Reply:
ah you're right 🤦‍♂, I haven't noticed that the condition is the same in both the cases. Thanks for the reply, let's leave it as it is.